`samples/snippets.py`: `test_create_instance_with_processing_units` flakes with 429

From [Samples - Python 3.7](https://source.cloud.google.com/results/invocations/a9753ac7-09f9-47e8-809f-61501c299dbe/targets/github%2Fpython-spanner%2Fsamples%2Fsamples/tests) and [Samples - Python 3.8](https://source.cloud.google.com/results/invocations/d86aa0fa-557a-4319-a247-8a229939bc44/targets/github%2Fpython-spanner%2Fsamples%2Fsamples/tests) today:

```

    def test_create_instance_with_processing_units(capsys):
        processing_units = 500
>       snippets.create_instance_with_processing_units(LCI_INSTANCE_ID, processing_units)

snippets_test.py:87:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
snippets.py:81: in create_instance_with_processing_units
    operation = instance.create()
../../google/cloud/spanner_v1/instance.py:322: in create
    metadata=metadata,
../../google/cloud/spanner_admin_instance_v1/services/instance_admin/client.py:825: in create_instance
    response = rpc(request, retry=retry, timeout=timeout, metadata=metadata,)
.nox/py-3-7/lib/python3.7/site-packages/google/api_core/gapic_v1/method.py:145: in __call__
    return wrapped_func(*args, **kwargs)
.nox/py-3-7/lib/python3.7/site-packages/google/api_core/grpc_helpers.py:69: in error_remapped_callable
    six.raise_from(exceptions.from_grpc_error(exc), exc)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

value = None
from_value = <_InactiveRpcError of RPC that terminated with:
	status = StatusCode.RESOURCE_EXHAUSTED
	details = "Quota exceeded for...uests per minute' of service 'spanner.googleapis.com' for consumer 'project_number:1012616486416'.","grpc_status":8}"
>

>   ???
E   google.api_core.exceptions.ResourceExhausted: 429 Quota exceeded for quota metric 'Instance create requests' and limit 'Instance create requests per minute' of service 'spanner.googleapis.com' for consumer 'project_number:1012616486416'.

<string>:3: ResourceExhausted
```

The fix would be to wrap the `instance.create` call with some form of exponential back-off, e.g. using the [`backoff`]() package:

```python
import backoff

w_backoff = backoff.on_exception(backoff.expo, exceptions.ResourceExhausted)

...

    def test_create_instance_with_processing_units(capsys):
        processing_units = 500
        w_backoff(snippets.create_instance_with_processing_units)(LCI_INSTANCE_ID, processing_units)
```


or the `RetryException` class from `google-cloud-testutils`.

--- a/samples/samples/snippets_test.py
+++ b/samples/samples/snippets_test.py
@@ -15,10 +15,12 @@
 import time
 import uuid
 
+from google.api_core import exceptions
 from google.cloud import spanner
 from google.cloud.spanner_v1.instance import Backup
 from google.cloud.spanner_v1.instance import Instance
 import pytest
+from test_utils.retry import RetryErrors
 
 import snippets
 
@@ -84,7 +86,10 @@ def test_create_instance(spanner_instance):
 
 def test_create_instance_with_processing_units(capsys):
     processing_units = 500
-    snippets.create_instance_with_processing_units(LCI_INSTANCE_ID, processing_units)
+    retry_429 = RetryErrors(exceptions.ResourceExhausted)
+    retry_429(enippets.create_instance_with_processing_units)(
+        LCI_INSTANCE_ID, processing_units,
+    )
     out, _ = capsys.readouterr()
     assert LCI_INSTANCE_ID in out
     assert "{} processing units".format(processing_units) in out

--- a/samples/samples/snippets_test.py
+++ b/samples/samples/snippets_test.py
@@ -87,7 +87,7 @@ def test_create_instance(spanner_instance):
 def test_create_instance_with_processing_units(capsys):
     processing_units = 500
     retry_429 = RetryErrors(exceptions.ResourceExhausted)
-    retry_429(enippets.create_instance_with_processing_units)(
+    retry_429(snippets.create_instance_with_processing_units)(
         LCI_INSTANCE_ID, processing_units,
     )
     out, _ = capsys.readouterr()

--- a/samples/samples/snippets_test.py
+++ b/samples/samples/snippets_test.py
@@ -86,7 +86,7 @@ def test_create_instance(spanner_instance):
 
 def test_create_instance_with_processing_units(capsys):
     processing_units = 500
-    retry_429 = RetryErrors(exceptions.ResourceExhausted)
+    retry_429 = RetryErrors(exceptions.ResourceExhausted, delay=15)
     retry_429(snippets.create_instance_with_processing_units)(
         LCI_INSTANCE_ID, processing_units,
     )

